ReactiveUI/PortableLibraryStubs.cs

```diff
@@ -10,7 +8,16 @@ public class PlatformRegistrations : IWantsToRegisterStuff
     {
         public void Register(Action<Func<object>, Type> registerFunction)
         {
+            if (IsRunningFromNUnit()) return;
             throw new Exception("You are referencing the Portable version of ReactiveUI in an App. Reference the platform-specific version.");
         }
+
+        private bool IsRunningFromNUnit()
+        {
+            var currentdomain = typeof(string).GetTypeInfo().Assembly.GetType("System.AppDomain").GetRuntimeProperty("CurrentDomain").GetMethod.Invoke(null, new object[] { });
```

Comment:
While it sounds like you're getting a bunch of feedback from the core team already, I thought I'd share some history on this.

In the early days, Splat did something not unlike what you're looking to bring in: https://github.com/paulcbetts/splat/blob/227ee96227df6fc542eea4c1b6eb1ffdcb7edd49/Splat/PlatformModeDetector.cs

This quickly devolved into supporting a lot of edge case scenarios - you can see all the test runners that were supported at the time there.

My advice here would be to wire up your own rules within the context of your app - it looks like you've got the rules sorted, it just needs to be wired up to Splat (and thus into ReactiveUI):

```csharp
public class NUnitModeDetector : IModeDetector
{
    // cache this computed value so that it's not re-evaluated
    Lazy<bool> checkForNUnit = new Lazy<bool>(IsRunningFromNUnit);
    public bool? InUnitTestRunner() { return checkForNUnit.Value; }
    public  bool? InDesignMode() { return null; }
    private bool IsRunningFromNUnit() {
        // this code goes here
    }
}

// elsewhere, in your ReactiveUI setup code
ModeDetector.OverrideModeDetector(new NUnitModeDetector());
```

Apologies if this code doesn't compile, but hopefully you get the idea...

cc @ghuntley remind me to document this somewhere when I'm back from holidays